packages/ubuntu_desktop_installer/test/connect_to_internet_page_test.dart

```diff
@@ -0,0 +1,193 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:provider/provider.dart';
+import 'package:ubuntu_desktop_installer/l10n.dart';
+import 'package:ubuntu_desktop_installer/pages/connect_to_internet/connect_model.dart';
+import 'package:ubuntu_desktop_installer/pages/connect_to_internet/connect_to_internet_model.dart';
+import 'package:ubuntu_desktop_installer/pages/connect_to_internet/connect_to_internet_page.dart';
+import 'package:ubuntu_desktop_installer/pages/connect_to_internet/ethernet_model.dart';
+import 'package:ubuntu_desktop_installer/pages/connect_to_internet/ethernet_view.dart';
+import 'package:ubuntu_desktop_installer/pages/connect_to_internet/wifi_model.dart';
+import 'package:ubuntu_desktop_installer/pages/connect_to_internet/wifi_view.dart';
+import 'package:ubuntu_desktop_installer/services.dart';
+import 'package:ubuntu_wizard/widgets.dart';
+
+import 'connect_to_internet_page_test.mocks.dart';
+
+@GenerateMocks([
+  ConnectToInternetModel,
+  EthernetModel,
+  EthernetDevice,
+  NetworkService,
+  UdevDeviceInfo,
+  UdevService,
+  WifiModel,
+  WifiDevice,
+])
+void main() {
+  setUpAll(() async {
+    await setupAppLocalizations();
+  });
+
+  Future<void> buildTestApp(
+    WidgetTester tester, {
+    required ConnectToInternetModel model,
+    bool? ethernetActive,
+    bool? wifiActive,
+  }) async {
+    final ethernetModel = MockEthernetModel();
+    when(ethernetModel.connectMode).thenReturn(ConnectMode.ethernet);
+    when(ethernetModel.devices).thenReturn([MockEthernetDevice()]);
+    when(ethernetModel.isActive).thenReturn(ethernetActive ?? false);
+    when(ethernetModel.canConnect).thenReturn(true);
+    when(ethernetModel.canContinue).thenReturn(true);
+    when(ethernetModel.isBusy).thenReturn(false);
+    when(ethernetModel.isEnabled).thenReturn(true);
+
+    final wifiDevice = MockWifiDevice();
+    when(wifiDevice.model).thenReturn('model');
+    when(wifiDevice.isAvailable).thenReturn(true);
+    when(wifiDevice.isBusy).thenReturn(false);
+    when(wifiDevice.scanning).thenReturn(false);
+    when(wifiDevice.accessPoints).thenReturn([]);
+
+    final wifiModel = MockWifiModel();
+    when(wifiModel.connectMode).thenReturn(ConnectMode.wifi);
+    when(wifiModel.devices).thenReturn([]);
```

Comment:
`wifiModel.devices` is later mocked to return `[wifiDevice]`, so this line is useless, right?